crates/kitsune_p2p/kitsune_p2p/src/gossip/sharded_gossip/bandwidth.rs

```diff
@@ -140,10 +139,23 @@ where
                             }
                             tokio::time::sleep(dur).await;
                         }
-                        governor::NegativeMultiDecision::InsufficientCapacity(_) => {
+                        governor::NegativeMultiDecision::InsufficientCapacity(mut cap) => {
                             tracing::error!(
-                                "Tried to send a message larger than the max message size"
+                                "Tried to send {} bits, which is larger than the maximum possible of {} bits. Allowing this large message through anyway!", bits, cap
                             );
+                            // TODO: rather than allowing this message through, we should bubble this error up so that the sender can split
+                            // the message into smaller chunks. We don't easily have that capacity right now, so, better to violate rate
+                            // limiting than to go into an infinite loop...
+
+                            // Drain the rate limiter's capacity completely, to be as accurate as possible.
+                            // (ideally we would just drain the capacity completely in one fell swoop, but `governor`'s API does not allow this.)
+                            while cap > 1 {
+                                outbound
+                                    .check_n(unsafe { NonZeroU32::new_unchecked(cap) })
+                                    .ok();
```

Comment:
why are we only doing this for outgoing and not incoming?

Reply:
Good point